documentation/src/docs/catalog.md

```diff
@@ -0,0 +1,161 @@
+
+[Pravega Schema Registry](https://github.com/pravega/schema-registry) is built for such purpose. It is the registry service built on Pravega that helps store and manage schemas for data stored in Pravega streams, and it also provides a factory of methods to standardize the serialization with built-in support for popular serialization formats in Avro, Protobuf, JSON schemas, as well as custom serialization.
+
+### `PravegaRegistryFormatFactory`
+
+The serialization specialized in the schema registry involves some extra control bits compared to a standard (de)serialization, so a new table format named `pravega-registry` is developed in the connector to define how to map binary data onto table columns on top of it.
+
+**Note:** It should only be used internally in the context of `PravegaCatalog` unless you understand what you are doing. Currently, it only supports Json and avro format without any additional encryption and compression codecs.
+
+It has following options:
+
+| Option                      | Required            | Default       | Type         | Description                                                                         |
+|-----------------------------|---------------------|---------------|--------------|-------------------------------------------------------------------------------------|
+| format                      | required            | (none)        | String       | Specify what format to use, here should be 'pravega-registry'                       |
+| pravega-registry.uri        | required            | (none)        | String       | Pravega Schema Registry service URI                                                 |
+| pravega-registry.namespace  | required            | (none)        | String       | Pravega Schema Registry namespace, should be the same name as Pravega scope         |
+| pravega-registry.group-id   | required            | (none)        | String       | Pravega Schema Registry group ID, should be the same name as Pravega stream         |
+| pravega-registry.format     | optional            | Avro          | String       | Default format for serialization in table sink, Valid values are 'Json' and 'Avro'  |
+| pravega-registry.json.*     | optional            | (none)        | -            | Specification for json format, completely inherited from official Flink Json format factory, refer to this [doc](https://ci.apache.org/projects/flink/flink-docs-stable/docs/connectors/table/formats/json/#format-options) for details                                  |
+
+Based on Pravega, its schema registry and this table format, a `PravegaCatalog` is built to manage Pravega streams as Flink tables.
+It can map all the streams with its Json/Avro schema registered, and users can directly read from/write to the stream without establishing connection with extra SQL DDL.
```

Comment:
```suggestion
It can map all the streams with its Json/Avro schema registered and users can directly read from/write to the stream without establishing connection with extra SQL DDL.
```